Formated with black

patka9797 opened the pull request.

patka9797 committed:
Formated with black

gortyy commented on repo_e/capital_letter.py:
```diff
@@ -0,0 +1,7 @@
+def capital(strings):
```
this function should take a list of strings and then capitalize every string in list, so if the input is `["raz", "DWA", "tRZY"]` the output should be `["Raz", "Dwa", "Trzy"]`

gortyy commented on repo_e/capital_letter.py:
```diff
@@ -0,0 +1,7 @@
+def capital(strings):
+    Capital_str=[]
+    for i, element in enumerate(strings):
+        if  element==element.capitalize():
```
please make a habit of formatting your code with black

patka9797 replied:
Ok

gortyy commented on repo_e/capital_letter.py:
```diff
@@ -0,0 +1,7 @@
+        if  element==element.capitalize():
+                Capital_str.append(element)
+    return Capital_str
+print(capital(['Krowa', 'swinia', 'Byk', 'Kun']))
```
try to write unit tests for you functions. You can use `pytest` for this. Create a directory called `tests` and there put your test files:
```sh
mkdir tests
touch tests/test_capital_letter.py
pytest
```
use the examples that I showed you when creating the recipes API as a reference

gortyy commented on repo_e/capital_letter.py:
```diff
@@ -0,0 +1,7 @@
+def capital(strings):
+    Capital_str=[]
```
please stick to snake_case naming, so `capital_str` instead of `Capital_str` (this is a mix of snake_case and CamelCase). Also - don't encode variable type in it's name, don't put `str`, `int`, `float`, `list` to names of variables

gortyy commented on repo_e/capital_letter.py:
```diff
@@ -0,0 +1,7 @@
+def capital(strings):
+    Capital_str=[]
+    for i, element in enumerate(strings):
```
no need for enumerate here, simple `for element in strings` would do the job

gortyy commented on repo_e/class_of_person.py:
```diff
@@ -0,0 +1,8 @@
+class Person:
+    def __init__(self, name, age):
```
please try to annotate arguments types:
```py
def __init__(self, name: str, age: int):
```
it will pay of

gortyy commented on repo_e/class_of_person.py:
```diff
@@ -0,0 +1,8 @@
+        self.age = age
+
+
+Person1 = ("patrycja", 25)
```
that's not how you create a Person object, do this instead:
```py
patrycja = Person("patrycja", 25)
```

gortyy commented on repo_e/class_of_person.py:
```diff
@@ -0,0 +1,8 @@
+
+
+Person1 = ("patrycja", 25)
+print("Witaj", Person1)
```
nope, please implement `greet` method inside person class instead

gortyy commented on repo_e/decorator_timer.py:
```diff
@@ -0,0 +1,10 @@
+
+
+
+@decorator
```
decorator is not defined in this file, did you forget to put it here?

patka9797 replied:
I didn't finish it

gortyy replied:
if you're stuck here just let me know so I can help

patka9797 replied:
I' m definietely stuck here

gortyy commented on repo_e/decorator_timer.py:
```diff
@@ -0,0 +1,10 @@
+from time import time
```
time is not used

patka9797 replied:
I know i didn't finish this file

gortyy commented on repo_e/fibonacci.py:
```diff
@@ -0,0 +1,19 @@
+def fibonacci():
```
it should be a infinite generator, so you don't ask the user for input but rather generate every fibonacci number using python's generators, something like that:
```py
def fibonacci():
  # logic for generating fibonacci numbers
  yield n1
```
note the `yield` keyword - that's what make an ordinary function a generator

gortyy commented on repo_e/is_divisible_3.py:
```diff
@@ -0,0 +1,2 @@
+list_ = [3, 5, 6, 9, 12, 333, 435345, 5435353]
+print(list(filter(lambda x: x % 3 == 0, list_)))
```
great! please wrap it in a function that accepts a list as argument

gortyy commented on repo_e/pair_of_numbers.py:
```diff
@@ -0,0 +1,10 @@
+def pair_of_numbers(list1, list2):
```
please try to come up with some better name, I know it's hard 😄

gortyy commented on repo_e/square_of_tuple.py:
```diff
@@ -0,0 +1,18 @@
+def square_tuple(tuple):
```
please try do it with list comprehension

patka9797 replied:
Done

gortyy commented on repo_e/square_of_tuple.py:
```diff
@@ -0,0 +1,18 @@
+def square_tuple(tuple):
+    square = ()
+    for number in tuple:
+        square += (number**2,)
```
that's not really a common approach, tuples are immutable (same as strings) and invoking `+=` on a tuple isn't a good practice

patka9797 replied:
I changed tuple into list

gortyy commented on repo_e/sum_od_even.py:
```diff
@@ -0,0 +1,9 @@
+def sum_of_even(integer):
```
nice 👍 please try to implement it using list comprehension and built-in `sum` function

patka9797 replied:
I made this

gortyy commented on repo_e/sum_words_string.py:
```diff
@@ -0,0 +1,12 @@
+def sum_word_in_string(string):
+    counts = dict()
+    str_split = str.split(string)
```
how will this function behave if you have following string:
```py
"raz, raz! raz. raz:"
```

gortyy replied:
I mean is it going to count "raz" four times?

gortyy commented on repo_e/sum_words_string.py:
```diff
@@ -0,0 +1,12 @@
+def sum_word_in_string(string):
+    counts = dict()
+    str_split = str.split(string)
+    for word in str_split:
```
same here, please try to name it better, str_split could be named `words` instead

gortyy commented on repo_e/sum_words_string.py:
```diff
@@ -0,0 +1,12 @@
+def sum_word_in_string(string):
```
overall, nice approach, you can also try to use `Counter` from `collections` library and see how it simplifies your solution

patka9797 replied:
I did it

gortyy commented on repo_e/unique_value.py:
```diff
@@ -0,0 +1,5 @@
+def unique(numbers):
+    return set(numbers)
```
works well but please return it as a list (right now you accept list and return set - you should accept list and return list)

gortyy reviewed (commented):
Overall - looks great, please test all of your solutions with pytest 🥇

patka9797 committed:
Update unique_value.py

Output list instead set

patka9797 committed:
Update sum_words_string.py

Trying Counter and rename varieable

patka9797 committed:
Update sum_od_even.py

Implement build sum()

patka9797 committed:
Update square_of_tuple.py

Changed  to list comprehension instead tuple

patka9797 committed:
Update pair_of_numbers.py

Rename list1 and list2 in better way

patka9797 committed:
Update is_divisible_3.py

Wraped into function with list as argument

patka9797 committed:
Update fibonacci.py

Try to create geberator with yield

patka9797 committed:
Update class_of_person.py

I'm trying to add greet in class but i'm not sure it will be good

patka9797 committed:
Add models and main to my app

patka9797 committed:
Merge branch 'New_b' of https://github.com/patka9797/repo_new into New_b

patka9797 committed:
Update class_of_person.py

Add greet method to class Person

patka9797 committed:
Delete coffee_app/app directory

Deleted app

gortyy commented on repo_e/sum_od_even.py:
```diff
@@ -0,0 +1,17 @@
+# print(sum_of_even([1,10,4,3]))
+
+def sum_of_even(integer):
+    Even=[]
+    for number in integer:
+        if number%2==0:
+            Even.append(number)
+    return sum(Even)
```
```suggestion
    return sum(number for number in numbers if number % 2 == 0)
```
`integer` is not best name for list of numbers, `numbers` would be better

gortyy commented on repo_e/sum_words_string.py:
```diff
@@ -0,0 +1,18 @@
+
+def sum_word_in_string(string):
+
+    print(Counter(string))
```
```suggestion
    return dict(Counter(string))
```

patka9797 committed:
Update sum_od_even.py

Change integer to numbers

patka9797 committed:
Update repo_e/sum_words_string.py

Co-authored-by: Dawid Śmiech <35574037+gortyy@users.noreply.github.com>

patka9797 committed:
Add files via upload

This is exercises from exercism. If you have any tips to make it more python give me it. Actually in this task all test went ok.

patka9797 committed:
Add files via upload

Program to gives random names for robots in factory. This program should also change name into other after reset robot software.

patka9797 committed:
Used time right now and add decorator

patka9797 committed:
formated with black

patka9797 committed:
Transcription DNA into RNA using dict structure
